### DISCUSSION TITLE
ITimeSpan conversion to float or int

### DISCUSSION THREAD

**Discussion by TravisWoolston** [melanchall/drywetmidi#291] [Q&A] [answered]
Docs: https://melanchall.github.io/drywetmidi/api/Melanchall.DryWetMidi.Interaction.TimeConverter.html

So I'm trying to use MIDI ticks to keep my scripts in sync with audio. (**if there is a better way to do this please let me know**). So I'd like to convert ITimeSpan to a float or int.

What I'm trying to do:
script for notes to move down screen.
public class MoveDown : MonoBehaviour
{
    public float fallRate = .01f;
    public VectorGrid VG;
    private SpriteRenderer spriteRenderer;
    public bool useVG;
    public long noteTime;
    public ITimeSpan currentTime;
    
    void Start()
    {
        spriteRenderer = GetComponent<SpriteRenderer>();
        if(useVG)
        VG = GameObject.FindWithTag("VG").GetComponent<VectorGrid>();
    }

    public void UpdatePosition(){
        float timeDifference = noteTime - currentTime;
        float offset = timeDifference * fallRate;
        transform.position = new Vector2(transform.position.x, transform.position.y - offset);
    }

**Comment by melanchall** [comment]
Hi, 

> So I'd like to convert ITimeSpan to a float or int.

What the type `ITimeSpan` is? What is float or int? MIDI ticks, seconds, what? Do you want to convert `currentTime` to this?

Please don't force me to pull the information from you. I have no time for this.

Thanks,
Max

**Reply by TravisWoolston** [reply, marked as answer]
Sorry if I wasn't clear what I'm trying to do. There is surprisingly little resources for this sort of thing so I appreciate what you've done here. I ended up just converting it to a string and then to a number. You're quite the one-man-band.

private float convertToFloat(ITimeSpan timeSpan){
    return Int32.Parse(Regex.Match(timeSpan.ToString(), @"\d+").Value);
}

**Reply by melanchall** [reply]
What about this?

```csharp
return ((MidiTimeSpan)timeSpan).TimeSpan;
```
